lib/pbench/server/utils.py

```diff
@@ -73,6 +75,11 @@ def quarantine(dest, logger, *files):
         if not os.path.exists(afile) and not os.path.islink(afile):
             continue
```

Comment:
What is this code supposed to be doing?  If the target file exists, then the conditional will short-circuit, and we won't worry about whether it's a symlink; If the target file does not exist, then clearly it's not a symlink, and so we don't need the second half of the conditional in that case, either.

(I realize, Dave, that you didn't change this code, but....)

Reply:
Ha; never even noticed, but you're right. Sometimes the original intent can be difficult to ascertain; in this case, most likely it's expecting that the input is an "action" link (we soft link the tarballs to `controller/TO-INDEX` subdirectories, for example, to tell the indexer to operate on them), and it intends to ignore if the file argument doesn't exist __or__ isn't a link. Although I'm not sure why it would assume the caller couldn't have passed the actual file path here.

This line, unfortunately, was coded by Chuck, 10 months ago; the guy whose abrupt departure opened the door for us to hire you, so asking might be awkward.

Reply:
Recorded in issue #2106 to resolve; I don't think this is as simple as `and` => `or`.